onnxruntime/core/mlas/lib/pooling.cpp

```diff
@@ -15,6 +15,21 @@ Module Name:
 --*/
 
 #include "mlasi.h"
+//
+// Define the prototype of the pooling kernel routine.
+//
+struct MLAS_WORK_BLOCK;
```

Comment:
I would prefer to get rid of the forward reference and just do "struct MLAS_WORK_BLOCK" instead. MLAS_WORK_BLOCK should be renamed MLAS_POOL_WORK_BLOCK too.

Reply:
But MLAS_POOL_WORK_BLOCK and MLAS_POOL_KERNEL_ROUTINE have a circular dependency

Reply:
I was suggesting doing the below to avoid the forward reference. This is more nit than real problem.

typedef
void
(MLAS_POOL_KERNEL_ROUTINE)(
    const struct MLAS_POOL_WORK_BLOCK* WorkBlock,
...